### PR TITLE
ENG-9286: Deprecate public docker route

### PR DESCRIPTION
**Pull request by gabfelp**
Update POST route permission so we can use only the private route for the new CPs

### CODE REVIEW

File: variables_sidecar.tf [hunk 1]
```diff
@@ -34,6 +34,12 @@ variable "control_plane" {
   type        = string
 }
 
+variable "external_http_port" {
+  description = "external CP HTTP port"
```
Comment (Yowgf) [inline variables_sidecar.tf:38]:
```suggestion
  description = "TCP/IP port of the control plane."
```

File: variables_sidecar.tf [hunk 2]
```diff
@@ -34,6 +34,12 @@ variable "control_plane" {
   type        = string
 }
 
+variable "external_http_port" {
```
Comment (Yowgf) [inline variables_sidecar.tf:37]:
```suggestion
variable "control_plane_port" {
```

File: variables_sidecar.tf [hunk 3]
```diff
@@ -34,6 +34,12 @@ variable "control_plane" {
   type        = string
 }
 
+variable "control_plane_port" {
+  description = "TCP/IP port of the control plane."
+  type        = string
+  default     = "8000"
```
Comment (wcmjunior) [inline variables_sidecar.tf:40]:
```suggestion
  default     = ""
```

File: variables_sidecar.tf [hunk 4]
```diff
@@ -34,6 +34,12 @@ variable "control_plane" {
   type        = string
 }
 
+variable "control_plane_port" {
+  description = "TCP/IP port of the control plane."
```
Comment (wcmjunior) [inline variables_sidecar.tf:38]:
```suggestion
  description = "Control plane API port."
```

File: ec2_resources.tf [hunk 5]
```diff
@@ -41,9 +41,16 @@ resource "aws_launch_configuration" "cyral-sidecar-lc" {
 
   echo "Downloading sidecar.compose.yaml..."
   function download_sidecar () {
-    local url="${local.protocol}://${var.control_plane}/deploy/sidecar.compose.yaml?TemplateVersion=${var.sidecar_version}&TemplateType=terraform&LogIntegration=${var.log_integration}&MetricsIntegration=${var.metrics_integration}&HCVaultIntegrationID=${var.hc_vault_integration_id}&WiresEnabled=${join(",", var.repositories_supported)}"
+    local url_token="${local.protocol}://${var.control_plane}:${var.control_plane_port}/v1/users/oidc/token"
+    local token=$(${local.curl} --fail --no-progress-meter --request POST "$url_token" -d grant_type=client_credentials -d client_id="${var.client_id}" -d client_secret="${var.client_secret}" 2>&1)
+    local token_error=$(echo $?)
+    if [[ $token_error -ne 0 ]]; then
+      return 1
+    fi
```
Comment (wcmjunior) [inline ec2_resources.tf:49]: Change this algorithm as follows:

```
if var.control_plane_port is defined {
  use var.control_plane_port in the curl command to get the token
} else {
  try curling on port 443
  if curl on port 443 fails {
    curl on port 8000
    if curl on port 8000 fails {
      fail the whole algorithm
    }
  }
}
```

File: ec2_resources.tf [hunk 6]
```diff
@@ -41,9 +41,32 @@ resource "aws_launch_configuration" "cyral-sidecar-lc" {
...
+      local token_error=$(echo $?)
+      if [[ $token_error -ne 0 ]]; then
+        return 1
+      fi
+    fi
+    local access_token=$(echo "$token" | jq -r .access_token)
+    local url="${local.protocol}://${var.control_plane}/deploy/docker-compose?TemplateVersion=${var.sidecar_version}&clientId=${var.client_id}&clientSecret=${var.client_secret}&SidecarId=${var.sidecar_id}&TemplateType=terraform&LogIntegration=${var.log_integration}&MetricsIntegration=${var.metrics_integration}&HCVaultIntegrationID=${var.hc_vault_integration_id}&WiresEnabled=${join(",", var.repositories_supported)}"
     echo "Trying to download the sidecar template from: $url"
-    if [[ $(${local.curl} -s -o /home/ec2-user/sidecar.compose.yaml -w "%%{http_code}" -L "$url") = 200 ]]; then
+    if [[ $(${local.curl} -s -o /home/ec2-user/sidecar.compose.yaml -w "%%{http_code}" -L "$url" -H "authorization: Bearer $access_token") = 200 ]]; then
```
Comment (wcmjunior) [inline ec2_resources.tf:69]:
```suggestion
    if [[ $(${local.curl} -s -o /home/ec2-user/sidecar.compose.yaml -w "%%{http_code}" -L "$url" -H "Authorization: Bearer $access_token") = 200 ]]; then
```

**Comment by Yowgf** [comment]
@gabfelp from the Terraform point of view this looks good to me :). Just remember to generate the README.md docs containing `control_plane_port`, using [terraform-docs](https://github.com/terraform-docs/terraform-docs), when the PR gets ready. You can do that for example by entering `terraform-docs markdown . >> temp.md` and then replacing the contents of README.md starting from `## Resources`.

**Review by wcmjunior — APPROVED** [review]

**Pull request merged**